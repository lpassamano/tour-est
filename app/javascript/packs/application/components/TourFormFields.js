import React, { Component } from "react";
import PropTypes from "prop-types";

export class TourFormFields extends Component {
  handleChange = event => {
    this.props.onChange(event);
  };

  render() {
    const {
      title,
      starting_point,
      directions,
      estimated_time,
      description
    } = this.props.tour;

    return (
      <div>
        <label htmlFor="title">Title: </label>
        <input
          name="title"
          type="text"
          value={this.props.title}
          onChange={this.handleChange}
        />
        <br />
        <label htmlFor="starting_point">Starting Point: </label>
        <input
          name="starting_point"
          type="text"
          value={this.props.starting_point}
          onChange={this.handleChange}
        />
        <br />
        <label htmlFor="directions">Directions: </label>
        <input
          name="directions"
          type="text"
          value={this.props.directions}
          onChange={this.handleChange}
        />
        <br />
        <label htmlFor="estimated_time">Estimated Time: </label>
        <input
          name="estimated_time"
          type="text"
          value={this.props.estimated_time}
          onChange={this.handleChange}
        />
        <br />
        <label htmlFor="description">Description: </label>
        <input
          name="description"
          type="text"
          value={this.props.description}
          onChange={this.handleChange}
        />
      </div>
    );
  }
}

TourFormFields.propTypes = {
  tour: PropTypes.shape({
    title: PropTypes.string,
    starting_point: PropTypes.string,
    directions: PropTypes.string,
    estimated_time: PropTypes.string,
    description: PropTypes.string
  }).isRequired,
  onChange: PropTypes.func.isRequired
};

export default TourFormFields;
