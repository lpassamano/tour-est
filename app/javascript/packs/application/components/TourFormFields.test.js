import { TourFormFields } from "./TourFormFields";
import React from "react";
import { shallow } from "enzyme";

describe("<TourFormFields />", () => {
  const setup = ({ onSubmit = jest.fn() }, initialValues = { title: "" }) => {
    const component = shallow(
      <TourFormFields onSubmit={onSubmit} initialValues={initialValues} />
    );
    return component;
  };

  const fillIn = (input, value) => {
    input.simulate("change", {
      target: { value, name: input.prop("name") }
    });
  };

  const submit = form => {
    form.simulate("submit", { preventDefault: () => null });
  };

  test("handleChange - when text is typed into the title field onChange is called", () => {
    const component = setup({});
    fillIn(component.find('[name="title"]'), "Best Tour Ever!");
    expect(component.state("title")).toEqual("Best Tour Ever!");
  });

  test("onSubmit - when button is clicked the current state is submitted", () => {
    const onSubmit = jest.fn();
    onSubmit.mockResolvedValue({ ok: true, data: { id: 1 } });
    const component = setup({ onSubmit });
    component.setState({ title: "New Tour" });
    submit(component.find("form"));

    expect(onSubmit).toHaveBeenCalledWith({
      title: "New Tour",
      description: "",
      directions: "",
      estimated_time: "",
      starting_point: ""
    });
  });

  test("Update Tour - when initialValues is passed as a prop the state is updated", () => {
    const initialValues = {
      id: 1,
      title: "Your Tour",
      description: "a tour",
      directions: "go to the place and make a left",
      estimated_time: "1 hr",
      starting_point: "the place"
    };
    const component = setup({}, initialValues);
    expect(component.state()).toEqual(initialValues);
  });

  test("render", () => {
    const component = setup({});
    expect(component).toMatchSnapshot();
  });
});
