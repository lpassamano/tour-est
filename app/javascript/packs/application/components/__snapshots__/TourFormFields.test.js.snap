// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TourFormFields /> render 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="title"
  >
    Title: 
  </label>
  <input
    name="title"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="starting_point"
  >
    Starting Point: 
  </label>
  <input
    name="starting_point"
    onChange={[Function]}
    placeholder="Tell your visitors where they should go to begin the tour."
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="directions"
  >
    Directions: 
  </label>
  <textarea
    name="directions"
    onChange={[Function]}
    placeholder="Let visitors know how to get to the starting point of the tour."
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="estimated_time"
  >
    Estimated Time: 
  </label>
  <input
    name="estimated_time"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <label
    htmlFor="description"
  >
    Description: 
  </label>
  <textarea
    name="description"
    onChange={[Function]}
    type="text"
    value=""
  />
  <br />
  <Button
    type="submit"
  >
    Save
  </Button>
</form>
`;
